resources/user_resource.py

```diff
@@ -0,0 +1,64 @@
+import redis
+import json
+from flask_restful import Resource, reqparse
+from passlib.hash import pbkdf2_sha256 as sha256
+from flask_jwt_extended import create_access_token, create_refresh_token
+from common.constants import USER_REGISTERED, USER_SIGN_IN_FAILED, USER_NOT_FOUND
+from database.database import db
+from models.user_model import User
+from flask import jsonify
+from flask_jwt_extended import jwt_required
+from schemas.user_schema import UserSignUpSchema, UserSignInSchema, UserDataViewSchema
+from flask_pydantic import validate
+
+
+redis_conn = redis.StrictRedis(host='localhost', port=6379, db=0)
+
+
+def hash_password(password):
+    return sha256.hash(password)
+
+
+def verify_password(password, hashed_password):
+    return sha256.verify(password, hashed_password)
+
+
+class UserSignUp(Resource):
+    @validate()
+    def post(self, body: UserSignUpSchema):
+        username = body.username
+        password = body.password
+        balance = body.balance
+        hashed_password = hash_password(password)
+        user = User(username=username, password=hashed_password, balance=balance)
```

Comment:
@haiderali358 why are you redefining variables so you can directly use them? And use meaningful naming instead of the body it should be user_detail or something like this.
```suggestion
        user = User(username=body.username, password=hashed_password, balance=body.balance)
```